packages/x-scheduler/src/internals/components/event-popover/content/FormContent.tsx

```diff
@@ -0,0 +1,434 @@
+    if (rawPlaceholder?.type !== 'creation') {
+      return;
+    }
+
+    const { start, end, surfaceType } = computeRange(adapter, next, nextIsAllDay);
+    const surfaceTypeToUse = rawPlaceholder.lockSurfaceType
+      ? rawPlaceholder.surfaceType
+      : surfaceType;
+
+    store.setOccurrencePlaceholder({
+      type: 'creation',
+      surfaceType: surfaceTypeToUse,
+      start,
+      end,
+      lockSurfaceType: rawPlaceholder.lockSurfaceType,
+    });
+  }
+
+  const handleChangeDateOrTimeField =
+    (field: keyof typeof when) =>
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      const value = event.currentTarget.value;
+      setErrors({});
+      setWhen((prev) => {
+        const next = { ...prev, [field]: value };
+        pushPlaceholder(next);
+        return next;
+      });
+    };
+
+  const handleToggleAllDay = (checked: boolean) => {
+    setIsAllDay(checked);
+    pushPlaceholder(when, checked);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const { start, end } = computeRange(adapter, when, isAllDay);
+
+    const form = new FormData(event.currentTarget);
+    const recurrenceValue = form.get('recurrence') as RecurringEventPresetKey;
+    const recurrenceModified =
+      defaultRecurrenceKey !== 'custom' && recurrenceValue !== defaultRecurrenceKey;
+    // TODO: This will change after implementing the custom recurrence editing tab.
+    const rrule =
+      recurrenceModified && recurrenceValue ? recurrencePresets[recurrenceValue] : undefined;
+
+    const resourceRawValue = form.get('resource');
+    const resourceValue =
+      resourceRawValue === '' ? undefined : (resourceRawValue as CalendarResourceId);
+
+    setErrors({});
+    const err = validateRange(adapter, start, end, isAllDay);
+    if (err) {
+      setErrors({ [err.field]: translations.startDateAfterEndDateError });
+      return;
+    }
+
+    const metaChanges = {
+      title: (form.get('title') as string).trim(),
+      description: (form.get('description') as string).trim(),
+      allDay: isAllDay,
+      resource: resourceValue,
+    };
+
+    if (rawPlaceholder?.type === 'creation') {
+      store.createEvent({ id: crypto.randomUUID(), ...metaChanges, start, end, rrule });
+    } else if (occurrence.rrule) {
+      const changes: CalendarEventUpdatedProperties = {
+        ...metaChanges,
+        id: occurrence.id,
+        start,
+        end,
+        ...(recurrenceModified ? { rrule } : {}),
+      };
+      store.updateRecurringEvent({
+        occurrenceStart: occurrence.start,
+        changes,
+        // TODO: Issue #19766 - Let the user choose the scope via UI.
+        scope: SCHEDULER_RECURRING_EDITING_SCOPE,
+      });
+    } else {
+      store.updateEvent({ id: occurrence.id, ...metaChanges, start, end, rrule });
+    }
+
+    onClose();
+  };
+
+  const handleDelete = useEventCallback(() => {
```

Comment:
The `useEventCallback` is probably not worth it since we pass it to a `<button />` and not a complex component